File: kafkaStreamsModelServer/src/main/scala/com/lightbend/scala/kafkastreams/store/store/custom/ModelStateStoreBuilder.scala
package com.lightbend.scala.kafkastreams.store.store.custom

import java.util
import java.util.{Objects}

import org.apache.kafka.streams.state.StoreBuilder

class ModelStateStoreBuilder(nm: String) extends StoreBuilder [ModelStateStore]{
  Objects.requireNonNull(nm, "name can't be null")

  var lConfig: util.Map[String, String] = new util.HashMap[String, String]
  var enableCaching: Boolean = false
  var enableLogging: Boolean = true

  override def build : ModelStateStore = new ModelStateStore(nm, enableLogging)

  override def withCachingEnabled: ModelStateStoreBuilder = {
    enableCaching = true
    this
  }

  override def withLoggingEnabled(config: util.Map[String, String]): ModelStateStoreBuilder = {
    Objects.requireNonNull(config, "config can't be null")
    enableLogging = true
    lConfig = config
    this
  }

  override def withLoggingDisabled: ModelStateStoreBuilder = {
    enableLogging = false
    lConfig.clear()
    this
  }

  override def logConfig: util.Map[String, String] = lConfig

  override def loggingEnabled: Boolean = enableLogging

  override def name: String = nm
}

File: kafkaStreamsModelServer/src/main/java/com/lightbend/java/kafkastreams/store/store/custom/ModelStateStoreBuilder.java
package com.lightbend.java.kafkastreams.store.store.custom;

import org.apache.kafka.streams.state.StoreBuilder;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class ModelStateStoreBuilder implements StoreBuilder<ModelStateStore> {

    private final String name;

    private Map<String, String> logConfig = new HashMap<>();
    boolean enableCaching;
    boolean enableLogging = true;


    public ModelStateStoreBuilder(final String name) {
        Objects.requireNonNull(name, "name can't be null");
        this.name = name;
    }

    @Override
    public ModelStateStore build() {
        return new ModelStateStore(name(), enableLogging);
    }

    @Override
    public  ModelStateStoreBuilder withCachingEnabled() {
        enableCaching = true;
        return this;
    }

    @Override
    public  ModelStateStoreBuilder withLoggingEnabled(final Map<String, String> config) {
        Objects.requireNonNull(config, "config can't be null");
        enableLogging = true;
        logConfig = config;
        return this;
    }

    @Override
    public  ModelStateStoreBuilder withLoggingDisabled() {
        enableLogging = false;
        logConfig.clear();
        return this;
    }

    @Override
    public Map<String, String> logConfig() {
        return logConfig;
    }

    @Override
    public boolean loggingEnabled() {
        return enableLogging;
    }

    @Override
    public String name() {
        return name;
    }

}
